### PR TITLE
Remove extension for Sinatra

### PR DESCRIPTION
**Pull request by epergo**
Contributes to https://github.com/sinatra/mustermann/issues/114

After the Sinatra extension has been extracted to its own gem/repository now we can remove the extension code from Mustermann itself.

### CODE REVIEW

**Comment by zzak** [comment]
I'm wondering what will be the experience if this code is simply removed, would it be better if we provided the same interface but raise an exception to the user telling them to add the extension gem?

**Comment by epergo** [comment]
> I'm wondering what will be the experience if this code is simply removed, would it be better if we provided the same interface but raise an exception to the user telling them to add the extension gem?

Right! The joy of deleting code blinded me and I didn't think about that :smile:  I can restore the [`extend_object`](
https://github.com/sinatra/mustermann/pull/116/files#diff-9205336053c6471b921892a24db663ecbc660b4d1815ae924f51a49998c4d0f9L120) method but instead of loading the removed code we can raise an exception with an explanatory message, would that work?

**Comment by namusyaka** [comment]
@epergo Thanks for your contribution.

I think you can simply put an exceptional message in `mustermann/lib/mustermann/extension.rb`, and then you can restore only `extend_object` without `register Extension`.
At least you should consider a case that `require`s `mustermann/extension`

**Comment by epergo** [comment]
> @epergo Thanks for your contribution.
> 
> I think you can simply put an exceptional message in `mustermann/lib/mustermann/extension.rb`, and then you can restore only `extend_object` without `register Extension`.
> At least you should consider a case that `require`s `mustermann/extension`

Updated in https://github.com/sinatra/mustermann/pull/116/commits/dcbf027f089dd07eaa546fcf66736100357c14b0 WDYT?

File: mustermann/lib/mustermann.rb [hunk 1]
```diff
@@ -120,7 +120,6 @@ def self.normalized_type(type)
   def self.extend_object(object)
```
Comment (zzak) [inline mustermann/lib/mustermann.rb:120]: Is this code called from anywhere? It seems we want to remove the `require` in this method 🤔

Comment (rkh) [reply mustermann/lib/mustermann.rb:120]: If I'm not mistaken, Sinatra's register method calls extend, which calls extend_object.

Comment (epergo) [reply mustermann/lib/mustermann.rb:120]: Correct it is used [here](https://github.com/sinatra/sinatra/blob/master/lib/sinatra/base.rb#L1454)

Comment (epergo) [reply mustermann/lib/mustermann.rb:120]: @zzak Do you want me to do any change? Maybe I missed the point.

Comment (zzak) [reply mustermann/lib/mustermann.rb:120]: @epergo Sorry, I was just wondering out loud nothing actionable -- but I think it'd be good to test this out a bit before we release anything

**Comment by olleolleolle** [comment]
@epergo Hi 👋 ! This is awesome, and shows a way forward for mustermann + 3.0. Does it still work? (GitHub Actions' logs are purged, swept away.)

**Comment by epergo** [comment]
> @epergo Hi 👋 ! This is awesome, and shows a way forward for mustermann + 3.0. Does it still work? (GitHub Actions' logs are purged, swept away.)

I've updated the PR with master, ruby 3.0 job still fails because of 1 test and the coverage report

**Pull request merged**